Libraries/LibWeb/Page/EventHandler.cpp

```diff
@@ -443,7 +443,8 @@ EventResult EventHandler::handle_mousewheel(CSSPixelPoint visual_viewport_positi
                 return EventResult::Dropped;
 
             auto page_offset = compute_mouse_event_page_offset(viewport_position);
-            auto offset = compute_mouse_event_offset(page_offset, *layout_node->first_paintable());
+            auto const& offset_paintable = layout_node->first_paintable() ? layout_node->first_paintable() : paintable.ptr();
+            auto offset = compute_mouse_event_offset(page_offset, *offset_paintable);
```

Comment:
Shouldn't such a logic live in `parent_element_for_event_dispatch` instead? I think we should skip over any parent element that has `display: contents`.